### ISSUE TITLE
"450 4.7.25 Client host rejected: cannot find your hostname" not matched

### ISSUE THREAD

**Issue by raubv0gel** [fail2ban/fail2ban#2442]
### Environment:

- Fail2Ban version: Fail2Ban v0.10.4
- OS version: Ubuntu 18.04.2 LTS (GNU/Linux 4.15.0-50-generic x86_64)
- Python version: 3.6.7
- [ ] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:

In `/var/log/mail.log` lines like
```
Jun  7 13:48:08 mail postfix/smtpd[9091]: NOQUEUE: reject: RCPT from unknown[103.58.16.46]: 450 4.7.25 Client host rejected: cannot find your hostname, [103.58.16.46]; from=<RichardRogers@logudorotours.it> to=<someone@mydomain.tld> proto=ESMTP helo=<logudorotours.it>
```
are not matched by `/etc/fail2ban/filter.d/postfix.conf` because of
```
^RCPT from [^[]*\[<HOST>\]%(_port)s: 45[04] 4\.7\.1 (?:Service unavailable\b|Client host rejected: cannot find your (reverse )?hostname\b)
```
only matches SMTP status code `45[04] 4\.7\.1` which excludes `450 4.7.25`.

#### Steps to reproduce

Create a file `/var/log/mail-test.log` and fill it with the log line above. After that, execute
```
fail2ban-regex /var/log/mail-test.log /etc/fail2ban/filter.d/postfix.conf /etc/fail2ban/filter.d/postfix.conf
```
that returns
```
Running tests
=============

Use   failregex filter file : postfix, basedir: /etc/fail2ban
Use      datepattern : Default Detectors
Use ignoreregex line : /etc/fail2ban/filter.d/postfix.conf
Use         log file : /var/log/mail-test.log
Use         encoding : UTF-8


Results
=======

Failregex: 0 total

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [1] {^LN-BEG}(?:DAY )?MON Day %k:Minute:Second(?:\.Microseconds)?(?: ExYear)?
`-

Lines: 1 lines, 0 ignored, 0 matched, 1 missed
[processed in 0.00 sec]

|- Missed line(s):
|  Jun  7 13:48:08 mail postfix/smtpd[9091]: NOQUEUE: reject: RCPT from unknown[103.58.16.46]: 450 4.7.25 Client host rejected: cannot find your hostname, [103.58.16.46]; from=<RichardRogers@logudorotours.it> to=<someone@mydomain.tld> proto=ESMTP helo=<logudorotours.it>
`-
```

#### Expected behavior

Should match those lines.

#### Observed behavior

Does not match those lines.

### Configuration, dump and another helpful excerpts

Default configuration.

**Comment by sebres** [comment]
Hmm... indeed, 4.7.x seems to have variate codes now.
Well, I don't see it as large problem to have `\d+` at end of status code here (and on other regex's with precise messages).
I'll fix it now.

**Comment by raubv0gel** [comment]
@sebres, you are great! Many thanks!

**Comment by sebres** [comment]
Fixed in 3b2f75414cb207aacb9e88409339dcbdfabc16ad.
